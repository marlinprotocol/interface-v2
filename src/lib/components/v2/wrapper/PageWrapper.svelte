<div class="flex items-center justify-center px-4 pb-4 pt-10 lg:px-0" data-testid="page-wrapper">
	<slot />
</div>
