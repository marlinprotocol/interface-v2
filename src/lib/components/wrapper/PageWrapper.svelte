<div
	class="flex max-h-[calc(100dvh-5rem)] w-full items-start justify-start overflow-auto px-7 py-8"
	data-testid="page-wrapper"
>
	<slot />
</div>
