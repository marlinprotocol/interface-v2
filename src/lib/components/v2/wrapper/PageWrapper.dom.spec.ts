import { cleanup, render } from '@testing-library/svelte';
import PageWrapper from './PageWrapper.svelte';
import html from '@playpilot/svelte-htm';

describe('PageWrapper', () => {
	afterEach(() => {
		cleanup();
		vi.resetAllMocks();
	});

	it('renders component successfully', () => {
		const { getByTestId, container } = render(PageWrapper);
		expect(getByTestId('page-wrapper')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it('renders with slot', () => {
		const { getByTestId } = render(
			html`<${PageWrapper}><div data-testid="test-page-wrapper">Test Page Wrapper</div></${PageWrapper}>`
		);
		expect(getByTestId('page-wrapper').contains(getByTestId('test-page-wrapper'))).toBe(true);
	});

	it('renders with correct slot', () => {
		const { getByTestId } = render(
			html`<${PageWrapper}><div data-testid="test-page-wrapper">Test Page Wrapper</div></${PageWrapper}>`
		);
		expect(getByTestId('page-wrapper').contains(getByTestId('test-page-wrapper'))).toBeTruthy();
		expect(getByTestId('test-page-wrapper').innerHTML === 'Test Page Wrapper').toBe(true);
	});
});
