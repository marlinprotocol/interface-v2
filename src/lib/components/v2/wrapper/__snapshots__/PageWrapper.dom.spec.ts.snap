// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageWrapper > renders component successfully 1`] = `
<body>
  <div>
    <div
      class="px-4 pb-4 pt-10 lg:px-0"
      data-testid="page-wrapper"
    />
    <!--&lt;PageWrapper&gt;-->
  </div>
</body>
`;
